client/src/CodeInput.js

```diff
@@ -20,6 +20,23 @@ class CodeInput extends React.Component {
         this.setState({lang: newLang});
     }
 
+    sendData() {
+        try {
+            const data = fetch('THE_ADDRESS', {
+                method: 'POST',
+                headers: {},
+                body: JSON.stringify({
+                    language: this.state.language,
+                    text: this.state.text
+                })
+            });
+            console.log(data);
+            this.setState({compiledCode: data});
```

Comment:
i'm very not used to passing data round in react and i'm unsure of the best way to get the request response data sent over to the codeoutput component. any good suggestions?

Reply:
Props are the way to go with this. I'll take a look at this section of code, and see what I might attempt. I had a lot of struggles with updating in different components and passing state around, so hopefully that'll help here. I'll workshop an idea before our meeting.